### PR TITLE
Change paths and os names to match corefx

### LINKED ISSUE #560: Match OS names with corefx
OS names should match corefx.  In addition, we should restrict parameters based on the allowable OSs.

Reference: https://github.com/dotnet/corefx/issues/1243

### PR DESCRIPTION
**Pull request by mmitche**
* binaries is now bin, intermediates is now obj
* modifying unixmscorlib build to instead be linuxmscorlib + osxmscorlib.
* Change OS=Unix to OSLinux/OS=OSX and modified properties to mimic corefx's TargetsUnix.

Fixes #560 and #559

### CODE REVIEW

File: build.cmd [hunk 1]
```diff
@@ -194,4 +195,6 @@ echo.
 echo BuildArch can be: x64
 echo BuildType can be: Debug, Release
 echo Clean - optional argument to force a clean build.
+echo linuxmscorlib - Build mscorlib for Linux
+echo osxmscorlib - Build mscorlib for osx
```
Comment (richlander) [inline build.cmd]: Nit - you might write "OS X".

**Comment by richlander** [comment]
For docs, it looks like the two changes are:

"binaries/intermediates/" -> "bin/obj/"
"unixmscorlib" -> "linuxmscorlib" | "osxmscorlib"

Is that right?

/cc @mmitche

**Comment by mmitche** [comment]
@richlander Yep

**Comment by mmitche** [comment]
@ellismg Fixed

**Comment by richlander** [comment]
I can make the changes or you. Just tell me when this merges if you want me to make the changes.

**Pull request merged**